### PR TITLE
Keep number field in focus when pressing dialpad buttons

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR prevents the dial pad text entry field from losing focus whenever a dial pad digit or the backspace button is pressed. This is useful as you can continue to use your keyboard to enter digits after you've clicked on a digit or the backspace button with the mouse.

Note that the in-call dial pad does not feature a backspace button.

Before:

before-focus

After:

Peek 2021-07-30 18-42

`Signed-off-by: Andrew Morgan <andrewm@element.io>`

Notes: Prevent the number field in the dial pad from losing keyboard focus when a digit or the backspace button is pressed.

Preview: https://61155ddddcb64a08fdbe311f--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by turt2live — APPROVED** [review]
is the dial pad easy to get access to as a regular user? If so, I'd like to see this run by [the accessibility community](https://matrix.to/#/#element-accessibility:matrix.org) to ensure it's not breaking any patterns they might have.

**Comment by anoadragon453** [comment]
The current plan is to test the accessibility this week as part of a general accessibility test for other parts of the app.

**Comment by t3chguy** [comment]
So there are major accessibility issues in the 1:1 voip area, not all related to this PR however

1. the buttons have no labels so as a blind user I have no idea which one is the keypad, they're all just "button"
2. the buttons vanish visually whilst the user's keyboard focus is on them
3. backspace button in dial pad missing label "button"
image
4. Not sure how great the auto-focus of the field is for keyboard accessibility, if I am trying to interact with the buttons, being sent to the input is jarring and means I have to tab a bunch of times to get back
5. Call button is missing any label "button"
image

**Comment by t3chguy** [comment]
probably one issue regarding 1:1 voip accessibility would be fine for the other points

File: src/components/views/voip/DialPad.tsx [hunk 1]
```diff
@@ -54,9 +54,9 @@ class DialPadButton extends React.PureComponent<IButtonProps> {
 }
 
 interface IProps {
-    onDigitPress: (string) => void;
+    onDigitPress: (string, object) => void;
     hasDial: boolean;
-    onDeletePress?: (string) => void;
+    onDeletePress?: (object) => void;
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/voip/DialPad.tsx:59]: Its still incorrect, that's the identifier, should be e.g `o: object`

**Comment by anoadragon453** [comment]
@t3chguy should hopefully have covered them all. Thanks for the review, obviously my typescript type game is still lacking!

**Review by t3chguy — APPROVED** [review]
:shipit:

**Pull request merged**